source/common/http/conn_manager_impl.cc

```diff
@@ -477,13 +477,64 @@ void ConnectionManagerImpl::chargeTracingStats(const Tracing::Reason& tracing_re
   }
 }
 
+// TODO(chaoqin-li1123): Make on demand vhds and on demand srds works at the same time.
 void ConnectionManagerImpl::RdsRouteConfigUpdateRequester::requestRouteConfigUpdate(
+    Http::RouteConfigUpdatedCallbackSharedPtr route_config_updated_cb) {
+  absl::optional<Router::ConfigConstSharedPtr> route_config = parent_.routeConfig();
+  Event::Dispatcher& thread_local_dispatcher =
+      parent_.connection_manager_.read_callbacks_->connection().dispatcher();
+  Router::ScopeKeyPtr scope_key;
+  if (route_config.has_value() && route_config.value()->usesVhds()) {
+    // On demand vhds
+    ASSERT(!parent_.filter_manager_.requestHeaders()->Host()->value().empty());
+    const auto& host_header =
+        absl::AsciiStrToLower(parent_.filter_manager_.requestHeaders()->getHostValue());
+    requestVhdsUpdate(host_header, thread_local_dispatcher, std::move(route_config_updated_cb));
+    return;
+  } else if (parent_.snapped_scoped_routes_config_ != nullptr &&
+             (scope_key = parent_.snapped_scoped_routes_config_->computeScopeKey(
+                  *parent_.filter_manager_.requestHeaders()))) {
```

Comment:
nit:put this into the curly bracket to look cleaner.

Reply:
Will do.